server/src/main/java/org/apache/cassandra/sidecar/routes/OperationalJobHandler.java

```diff
@@ -0,0 +1,120 @@
+
+    @Inject
+    public OperationalJobHandler(InstanceMetadataFetcher metadataFetcher,
+                                 ExecutorPools executorPools,
+                                 CassandraInputValidator validator,
+                                 OperationalJobManager jobManager)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.jobManager = jobManager;
+    }
+
+    @Override
+    protected Void extractParamsOrThrow(RoutingContext context)
+    {
+        return null;
+    }
+
+    @Override
+    public void handleInternal(RoutingContext context, HttpServerRequest httpRequest, String host, SocketAddress remoteAddress, Void request)
+    {
+        UUID jobId = validateJobIdParam(context);
+        executorPools.service()
+                     .executeBlocking(() -> {
+                         OperationalJob job = jobManager.getJobIfExists(jobId);
+                         if (job == null)
+                         {
+                             logger.info("No operational job found with the jobId. jobId={}", jobId);
+                             throw wrapHttpException(HttpResponseStatus.NOT_FOUND,
+                                                     String.format("Unknown job with ID: %s. Please retry the operation.", jobId));
+                         }
+                         return job;
+                     })
+                     .onFailure(cause -> processFailure(cause, context, host, remoteAddress, request))
+                     .onSuccess(job -> sendStatusBasedResponse(context, jobId, job));
+    }
+
+    private UUID validateJobIdParam(RoutingContext context)
+    {
+        String requestJobId = context.pathParam(OPERATIONAL_JOB_ID_PATH_PARAM.substring(1));
+        if (requestJobId == null)
+        {
+            throw wrapHttpException(HttpResponseStatus.BAD_REQUEST,
+                                    OPERATIONAL_JOB_ID_PATH_PARAM + " is required but not supplied");
+        }
+
+        UUID jobId;
+        try
+        {
+            jobId = UUID.fromString(requestJobId);
+        }
+        catch (IllegalArgumentException e)
+        {
+            logger.info("Invalid jobId. jobId={}", requestJobId);
+            throw wrapHttpException(HttpResponseStatus.BAD_REQUEST, String.format("Invalid job ID provided :%s.", requestJobId));
+        }
+        return jobId;
+    }
+
+    public void sendStatusBasedResponse(RoutingContext context, UUID jobId, OperationalJob job)
+    {
+        OperationalJobStatus status = job.status();
+        String reason = null;
+        switch(status)
+        {
+            case SUCCEEDED:
+                context.response().setStatusCode(HttpResponseStatus.OK.code()); break;
+            case CREATED:
+            case RUNNING:
+                context.response().setStatusCode(HttpResponseStatus.ACCEPTED.code()); break;
+            case FAILED:
+                reason = job.asyncResult().cause().getMessage();
```

Comment:
do we need to send a status code here?

Reply:
Added

Reply:
does 200 make sense here?

Reply:
Yes, since 200 is the status of the (current) job API request, which succeeded. The response body has the status of the job being queried which is `FAILED`.

Reply:
yep. 200 to indicate the request is successful. The status of the job is expressed in the response body.